File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/BlueBook.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class BlueBook extends Equipable {
	{
		name = Locale.get("blue_book");
		description = Locale.get("blue_book_desc");
		sprite = "item-less_mana_used_but_minus_hp_percent";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);

		if (this.owner instanceof Player) {
			Player player = (Player) this.owner;

			player.setHpMax(player.getHpMax() - 4);
			player.manaModifier -= 0.5;
		}
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);

		if (this.owner instanceof Player) {
			Player player = (Player) this.owner;


			player.setHpMax(player.getHpMax() + 4);
			player.manaModifier += 0.5;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/BlueBook.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equipable

import org.rexcellentgames.burningknight.assets.Locale

class BlueBook : Equipable() {
	init {
		name = Locale.get("blue_book")
		description = Locale.get("blue_book_desc")
		sprite = "item-less_mana_used_but_minus_hp_percent"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.hpMax = this.owner.hpMax - 4
		this.owner.manaModifier -= 0.5f
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.hpMax = this.owner.hpMax + 4
		this.owner.manaModifier += 0.5f
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}
}